#! /bin/sh /usr/share/dpatch/dpatch-run
## 11_memleak.dpatch by Stefan Huelswitt <s.huelswitt@gmx.de> and Daniel Thompson <daniel.thompson@st.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Kill some memory leaks.

@DPATCH@
diff -urNad vdr-1.3.23/dvbplayer.c /tmp/dpep.HpFRhI/vdr-1.3.23/dvbplayer.c
--- vdr-1.3.23/dvbplayer.c	2005-04-12 01:28:38.278587027 +0100
+++ /tmp/dpep.HpFRhI/vdr-1.3.23/dvbplayer.c	2005-04-12 01:28:38.569563448 +0100
@@ -296,6 +296,7 @@
      nonBlockingFileReader->Clear();
   if ((readIndex = backTrace->Get(playDir == pdForward)) < 0)
      readIndex = writeIndex;
+  delete readFrame;
   readFrame = NULL;
   playFrame = NULL;
   ringBuffer->Clear();
diff -urNad vdr-1.3.23/epg.c /tmp/dpep.HpFRhI/vdr-1.3.23/epg.c
--- vdr-1.3.23/epg.c	2005-04-12 01:28:38.241590025 +0100
+++ /tmp/dpep.HpFRhI/vdr-1.3.23/epg.c	2005-04-12 01:29:20.344178196 +0100
@@ -29,9 +29,7 @@
 {
   unsigned int Stream, Type;
   int n = sscanf(s, "%X %02X %3c %a[^\n]", &Stream, &Type, language, &description);
-  if (n != 4)
-     description = NULL;
-  else if (isempty(description)) {
+  if (n != 4 || isempty(description)) {
      free(description);
      description = NULL;
      }
